File: Chapter2/2.3/2.3.4-The 3n + 1 problem/2.3.4-The 3n + 1 problem.cpp
#include <iostream>
using namespace std;

int main(){
    int i,j,ans,i_,j_;
    while (cin>>i>>j){
        ans=0;
        i_=min(i,j),j_=max(i,j);
        for (int u=i_;u<=j_;u++){
            int count=0;
            int temp = u;
            while (temp!=1){
                count+=1;
                if (temp%2==1){
                    temp=3*temp+1;
                }
                else{
                    temp=temp/2;
                }
            }
            count+=1;
            if (count>ans){
                    ans=count;
                }
        }
            cout<<i<<" "<<j<<" "<<ans<<endl;
    }
}

File: Chapter2/2.3/2.3.4-The 3n + 1 problem/2.3.4-The 3n + 1 problem.py
#超时
while True:
    try:
        i,j=map(int,input().split())
        if i>j:
            i,j=j,i
            ex=True
        else:
            ex=False
        ans=0
        for u in range(i,j+1):
            count=1#包含print
            while u!=1:
                count+=1
                if u%2==1:
                    u=3*u+1
                else:
                    u=u/2
            if count>ans:
                ans=count
        if ex==True:
            i,j=j,i
        print(f'{i} {j} {ans}')
    except:
        break